wagtail/contrib/forms/models.py

```diff
@@ -115,6 +119,61 @@ def clean_name(self):
         FieldPanel('default_value', classname="formbuilder-default"),
     ]
 
+
+    def save(self, *args, **kwargs):
+        """
+        When new fields are created, generate a template safe ascii name to use as the
+        JSON storage reference for this field. Previously created fields will be updated
+        to use the legacy unidecode method via checks & _migrate_legacy_clean_name.
+        """
+
+        is_new = self.pk is None
+        if is_new:
+            clean_name = get_field_clean_name(self.label)
+            self.clean_name = clean_name
+
+        super().save(*args, **kwargs)
+
+
+    @classmethod
+    def _migrate_legacy_clean_name(cls):
+        """
+        Ensure that existing data stored will be accessible via the legacy clean_name.
+        When checks run, replace any blank clean_name values with the unidecode conversion.
+        """
+
+        try:
+            objects = cls.objects.filter(clean_name__exact='')
+
+            for obj in objects:
+                legacy_clean_name = str(slugify(string_to_ascii(obj.label)))
```

Comment:
I'm concerned about the use of `string_to_ascii` here, as we're not distinguishing between uses that rely on it preserving unidecode semantics (i.e. this one) and ones that just want an ASCII string and don't care how (i.e. the HTTP headers for document serving) - I think it'd better to use `unidecode` directly here. If we do it as follows, I think we can avoid having to revisit this code later when we drop unidecode:

* if `cls.objects.filter(...)` returns no results, return immediately
* Otherwise, import unidecode in a try/catch
* If that throws an ImportError, rethrow it with an explanatory message, something like: "You have form submission data that was created on an older version of Wagtail and requires the unidecode library to retrieve it correctly. Please install the unidecode package." (Currently this should never happen on a properly installed Wagtail instance, while unidecode is listed as a dependency in setup.py, but this will come into play when we drop that dependency, for anyone who upgrades directly from 2.9 to that version.)
* Otherwise, proceed to populate clean_name, using `unidecode` to convert the label

At that point, I think it would be reasonable to drop `string_to_ascii` entirely and use `safe_snake_case` in the remaining spots instead. Then, the only thing left to do later (in 2.12, if we follow the usual deprecation timeline) is to drop the unidecode dependency from setup.py.